src/app/store/trips/effects.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+
+import { switchMap } from 'rxjs';
+import { Trip } from '../../models/trips';
+import { ApiService } from '../../services/api.service';
+import {
+  createNewTrip,
+  createNewTripComplete,
+  deleteTrip,
+  deleteTripComplete,
+  getAllTrips,
+  getAllTripsComplete,
+  updateTrip,
+  updateTripComplete,
+} from './actions';
+
+@Injectable()
+export class TripsEffects {
+  getAllTrips$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(getAllTrips.type),
+      switchMap(() =>
+        this.apiService
+          .getTrips()
+          .then(data => {
+            if (data.empty) {
+              return getAllTripsComplete({ trips: [] });
+            }
+            const trips = data.docs.map(doc => {
+              return {
+                ...(doc.data() as Trip),
+                docId: doc.id,
+              };
+            });
+            return getAllTripsComplete({ trips: trips });
+          })
+          .catch(err => {
+            console.error(err);
+            return getAllTripsComplete({ trips: [] });
```

Comment:
here as well, unless you have specific UI that maybe checks if `trips.length === 0` for displaying some sort of notice, you can omit these an just have it check for `null` or `undefined` as well. Let me know if there is a specific reason you are explicitly setting it to an empty array

Reply:
The UI will eventually check if the length is bigger than 0 yes. I just don't have a UI for now really so a few things are still subject to change.